### PR TITLE
Component search tests, 1st set

### PR DESCRIPTION
**Pull request by tisnik**
This is the 1st set of test for the updated component and component-search API. More complicated tests will follow after we will get more info about the expected behaior ;)

### CODE REVIEW

**Comment by fridex** [comment]
@samuzzal-choudhury, could you please check this not to get it rotten. Thanks!

File: integration-tests/features/components.feature [hunk 1]
```diff
@@ -0,0 +1,41 @@
+Feature: Components API V1
+
+  Scenario: Check the API entry point
+    Given System is running
+    When I access /api/v1/
+    Then I should get 200 status code
+
```
Comment (samuzzal-choudhury) [inline integration-tests/features/components.feature]: @tisnik we will have to check the authentication module as well. Could you add steps for checking error 401 as well for /component-search and /component-analyses?

(Here is the location](https://github.com/fabric8-analytics/fabric8-analytics-server/tree/master/hack/auth_test_fixtures) of test auth token generation code. We can reuse this code to generate the Auth token for calling the API methods.

File: integration-tests/features/components.feature [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    When I access /api/v1/component-search/
+    Then I should get 404 status code
+
+  Scenario: Check if search for packages handle empty results
+    Given System is running
+    Given Component search service is running
+    When I search for component foobar
+    Then I should see 0 components
+
+  Scenario: Check that component analyses endpoint checks if all resources are specified
```
Comment (samuzzal-choudhury) [inline integration-tests/features/components.feature]: We should have a success scenario checked as well for /component-analyses.

File: integration-tests/features/components.feature [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    Given System is running
+    Given Component search service is running
+    When I access /api/v1/component-analyses
+    Then I should get 404 status code
+    When I access /api/v1/component-analyses/npm
+    Then I should get 404 status code
+    When I access /api/v1/component-analyses/component
+    Then I should get 404 status code
```
Comment (samuzzal-choudhury) [inline integration-tests/features/components.feature]: Bookkeeping is supported for both /component-analyses and /stack-analyses-v2 currently. Please include that as well in the tests. API_RESULTS in the new table. the user_profile hash can be found by decoding the auth token which can then be compared with the bookkept user_profile_digest.

Note: We have to make sure to have only one entry in the api_requests table for api_type "component_analyses" before testing the scenario.

**Comment by tisnik** [comment]
[test this]

**Comment by tisnik** [comment]
[test this]

**Comment by samuzzal-choudhury** [comment]
[test this]

**Comment by samuzzal-choudhury** [comment]
Merging without considering the "Checks Failed" as we need integration test scenarios to be available on staging ASAP.

**Pull request merged**